src/app/icd/client/RefreshKeySender.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "RefreshKeySender.h"
+#include "CheckInDelegate.h"
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandPathParams.h>
+#include <app/InteractionModelEngine.h>
+#include <app/OperationalSessionSetup.h>
+#include <memory>
+
+namespace chip {
+namespace app {
+
+RefreshKeySender::RefreshKeySender(CheckInDelegate * apCheckInDelegate, ICDClientInfo & aICDClientInfo,
+                                   ICDClientStorage * aICDClientStorage) :
+    mOnConnectedCallback(HandleDeviceConnected, this),
+    mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{
+    mpCheckInDelegate  = apCheckInDelegate;
+    mICDClientInfo     = aICDClientInfo;
+    mpICDClientStorage = aICDClientStorage;
```

Comment:
Why are these not just initializers too?  They should be.

Reply:
Updated.